pkg/apis/autoscaling/register.go

```diff
@@ -52,8 +52,80 @@ const (
 	// TargetAnnotationKey is the annotation to specify what metric value the
 	// PodAutoscaler should attempt to maintain. For example,
 	//   autoscaling.knative.dev/metric: cpu
-	//   autoscaling.knative.dev/target: 75   # target 75% cpu utilization
+	//   autoscaling.knative.dev/target: "75"   # target 75% cpu utilization
 	TargetAnnotationKey = GroupName + "/target"
+	// TargetMin is the minimum allowable target. Values less than
+	// zero don't make sense.
+	TargetMin = 1
+
+	// WindowAnnotationKey is the annotation to specify the time
+	// interval over which to calculate the average metric.  Larger
+	// values result in more smoothing. For example,
+	//   autoscaling.knative.dev/metric: concurrency
+	//   autoscaling.knative.dev/window: "2m"
+	// Only the kpa.autoscaling.knative.dev class autoscaler supports
+	// the window annotation.
+	WindowAnnotationKey = GroupName + "/window"
+	// WindowMinSeconds is the minimum allowable stable autoscaling
+	// window. KPA-class autoscalers calculate the desired replica
+	// count every 2 seconds (tick-interval in config-autoscaler) so
+	// the closer the window gets to that value, the more likely data
+	// points will be missed entirely by the panic window which is
+	// smaller than the stable window. Anything less than 6 second
+	// isn't going to work well.
+	WindowMinSeconds = 6
```

Comment:
Should this be a `time.Duration` vs `int`?

Reply:
Yes.